protocol_details/rpmsg_protocol.rst

```diff
@@ -0,0 +1,89 @@
+
+.. image:: ../images/rpmsg_header.jpg
+
+Special consideration should be taken if an alignment greater than 16 bytes is required; however, this is not typical for a shared memory, which should be fast and is therefore often not cached (alignment greater than 8 bytes is not needed at all).
+
+Flags Field
+~~~~~~~~~~~
+
+The flags field of the RPMsg header is currently unused by RPMsg and is reserved. Any propositions for what this field could be used for is welcome. It could be released for application use, but this can be considered as inconsistent - RPMsg header would not be aligned and the reserved field would be therefore useless.
+
+.. image:: ../images/rpmsg_flags.jpg
+
+RPMsg Channel
+~~~~~~~~~~~~~
+
+Every remote core in RPMsg component is represented by RPMsg device that provides a communication channel between master and remote, hence RPMsg devices are also known as channels RPMsg channel is identified by the textual name and local (source) and destination address. The RPMsg framework keeps track of channels using their names.
+
+.. _rpmsg-endpoint:
+
+RPMsg Endpoint
+~~~~~~~~~~~~~~
+
+RPMsg endpoints provide logical connections on top of RPMsg channel. It allows the user to bind multiple rx callbacks on the same channel.
+
+Every RPMsg endpoint has a unique src address and associated call back function. When an application creates an endpoint with the local address, all the further inbound messages with the destination address equal to local address of endpoint are routed to that callback function. Every channel has a default endpoint which enables applications to communicate without even creating new endpoints.
+
+.. image:: ../images/rpmsg_endpoint.jpg
+
+**************************
+RPMsg Protocol Limitations
+**************************
+
+    The RPMSG document has the concept of the static channel but it is not implemented in upstream Linux and OpenAMP. Please see https://www.kernel.org/doc/Documentation/rpmsg.txt. The protocol must define connection sequence when channel is created statically.
+    No synchronization point is defined by the RPMsg after which both sides can communicate reliably with each other. In the current protocol, at startup, the master sends notification to remote to let it know that it can receive name service announcement. However, master does not consider the fact that if the remote is ready to handle notification at this point in time.
```

Comment:
s/by the RPMsg/by the RPMsg framework/
s/master/the host/ where needed.
s/sends notification/sends a notification/
s/service announcement/service announcements/